sqlx-core/src/sqlite/options/mod.rs

```diff
@@ -458,4 +466,40 @@ impl SqliteConnectOptions {
             .insert(extension_name.into(), Some(entry_point.into()));
         self
     }
+
+    /// Execute `PRAGMA optimize;` on the SQLite connection before closing.
+    ///
+    /// The SQLite manual recommends using this for long-lived databases.
+    ///
+    /// This will collect and store statistics about the layout of data in your tables to help the query planner make better decisions.
+    /// Over the connection's lifetime, the query planner will make notes about which tables could use up-to-date statistics so this
+    /// command doesn't have to scan the whole database every time. Thus, the best time to execute this is on connection close.
+    ///
+    /// You may also wish to set the value for [`analysis_limit`][Self::analysis_limit] so database connections close quickly,
+    /// even if your database is very large. Set `limit` parameter if you wish to override [`analysis_limit`][Self::analysis_limit]
+    /// before execute `PRAGMA optimize;`.
+    ///
+    /// Not enabled by default.
+    ///
+    /// See [the SQLite manual](https://www.sqlite.org/lang_analyze.html#automatically_running_analyze) for details.
+    pub fn optimize_on_close(mut self, enabled: bool, limit: impl Into<Option<u32>>) -> Self {
+        self.optimize_on_close = enabled;
+        self.analysis_limit = limit.into();
```

Comment:
I would suggest renaming this field to something like `optimize_on_close_analysis_limit` to make it clearer in the code what it affects.

Technically, the more idiomatic API design here would be to combine these two options into an enum, since setting the limit while `enabled: false` does nothing: 

```rust
pub enum OptimizeOnClose {
    Enabled { analysis_limit: Option<u32> },
    Disabled
}
```

However, I'm on the fence about that, as that's another thing for the user to import. Using something like that internally may not be a bad idea, though.